ape_etherscan/client.py

```diff
@@ -79,10 +83,64 @@ def __init__(self, network_name: str, address: str):
         self._address = address
         super().__init__(network_name, "contract")
 
-    def get_source_code(self) -> Optional[Dict]:
+    def get_source_code(self) -> SourceCodeResponse:
         params = {**self.base_params, "action": "getsourcecode", "address": self._address}
+        result = self._get(params=params) or []
+        return SourceCodeResponse(**result[0]) if len(result) == 1 else SourceCodeResponse()
+
+
+class AccountClient(_APIClient):
+    def __init__(self, network_name: str, address: str):
+        self._address = address
+        super().__init__(network_name, "account")
+
+    def get_all_normal_transactions(
+        self,
+        start_block: Optional[int] = None,
+        end_block: Optional[int] = None,
+        offset: int = 100,
+        sort: str = "asc",
+    ) -> Iterator[Receipt]:
+        page_num = 1
+        last_page_results = offset  # Start at offset to trigger iteration
+        while last_page_results == offset:
+            page = self._get_page_of_normal_transactions(
+                page_num, start_block, end_block, offset=offset, sort=sort
+            )
+
+            if len(page):
+                for receipt_data in page:
+                    if "confirmations" in receipt_data:
+                        receipt_data["required_confirmations"] = receipt_data.pop("confirmations")
+                    if "txreceipt_status" in receipt_data:
+                        receipt_data["status"] = receipt_data.pop("txreceipt_status")
+
+                    receipt: Receipt = Receipt.decode(receipt_data)  # type: ignore
+                    yield receipt
```

Comment:
You can do this iteration in `ExplorerAPI.get_account_transactions` I was just saying do the paging loop unroll here because it's where the paging is being handled already.

splitting up the two for loops makes both functions easier to read.

Reply:
Okay, that means we either have to return `dict` from this class or make a whole new receipt data class just for etherscan

Reply:
I tried making a `TransactionResponse` but it just seems weird because we have to go from `Dict` -> `TransactionResponse` -> `Dict` -> `Receipt` (via `receipt_class.decode()`, so it seems better to just go straight from `Dict` -> `Receipt`, like we do in the `ProviderAPI` via making calls to `web3`. Like, we are not making models around the responses from `web3` either.

Reply:
also fair